runelite-client/src/main/java/net/runelite/client/plugins/statusbars/BarRenderer.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2019, Jos <Malevolentdev@gmail.com> & Rheon <https://github.com/Rheon-D>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *     list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *     this list of conditions and the following disclaimer in the documentation
+ *     and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+package net.runelite.client.plugins.statusbars;
+
+import lombok.RequiredArgsConstructor;
+import net.runelite.client.ui.FontManager;
+import java.awt.Color;
+import java.awt.Graphics2D;
+import java.awt.Image;
+import java.util.function.Supplier;
+
+@RequiredArgsConstructor
+class BarRenderer
+{
+	private static final Color BACKGROUND = new Color(0, 0, 0, 150);
+	private static final Color OVERHEAL_COLOR = new Color(216, 255, 139, 150);
+	private static final int OVERHEAL_OFFSET = 2;
+	private static final int HEAL_OFFSET = 3;
+	private static final int ICON_AND_COUNTER_OFFSET_X = 1;
+	private static final int ICON_AND_COUNTER_OFFSET_Y = 21;
+	private static final int SKILL_ICON_HEIGHT = 35;
+	private static final int COUNTER_ICON_HEIGHT = 18;
+	private static final int OFFSET = 2;
+	private static final int WIDTH = 20;
+	private static final int PADDING = 1;
+	private final StatusBarsConfig config;
+	private final Supplier<Integer> maxValueSupplier;
+	private final Supplier<Integer> currentValueSupplier;
+	private final Supplier<Integer> healSupplier;
+	private final Supplier<Color> colorSupplier;
+	private final Supplier<Color> healColorSupplier;
+	private final Supplier<Image> iconSupplier;
+	private int maxValue;
+	private int currentValue;
+	private final static int LEFT = 1;
+
+	private void refreshSkills()
+	{
+		maxValue = maxValueSupplier.get();
+		currentValue = currentValueSupplier.get();
+	}
+
+	/**
+	 * renderBar is called to render the bar, restorationbars, icons and counters on the gamescreen
```

Comment:
This should be written using indicative mood.
```suggestion
	 * Renders a status bar along with its restoration bar(s), icons and counters as appropriate
```